app/models/bookmark.rb

```diff
@@ -106,6 +170,14 @@ def bookmark_limit_not_reached
     )
   end
 
+  def valid_bookmarkable_type
+    return if !SiteSetting.use_polymorphic_bookmarks
+    return if Bookmark.valid_bookmarkable_types.include?(self.bookmarkable_type)
+    return if ["Post", "Topic"].include?(self.bookmarkable_type)
```

Comment:
I wonder if it makes sense to move `["Post", "Topic"]` to a constant. I see it being used in two places right now.

Reply:
Yep makes sense, they are the special snowflake types.